packages/podlet-js/src/compose/compose.ts

```diff
@@ -0,0 +1,102 @@
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+import type { ComposeSpecification, DefinitionsService, PropertiesServices } from 'compose-spec-ts';
+import { load, dump } from 'js-yaml';
+import type { PodContainer, PodContainerPort, PodmanPod } from './models/podman-pod';
+
+export const COMPOSE_SPECIFICATION_SUPPORTED: Set<keyof ComposeSpecification> = new Set([
+  'services',
+  'name',
+  'version',
+]);
+
+export const SERVICE_SUPPORTED: Set<keyof DefinitionsService> = new Set(['image', 'ports']);
+
+export class Compose {
+  #spec: ComposeSpecification;
+
+  protected constructor(spec: ComposeSpecification) {
+    this.#spec = spec;
+  }
+
+  public static fromString(raw: string): Compose {
+    return new Compose(load(raw) as ComposeSpecification);
+  }
+
+  public getServices(): PropertiesServices {
+    return this.#spec.services ?? {};
+  }
+
+  // todo: move to dedicated file
+  private toPodContainer([name, service]: [string, DefinitionsService]): PodContainer {
+    Object.keys(service).forEach((key: string) => {
+      if (!SERVICE_SUPPORTED.has(key)) throw new Error(`unsupported option ${key} for service ${name}`);
+    });
+
+    if (!service.image) throw new Error('missing image');
+
+    const ports: Array<PodContainerPort> | undefined = service.ports?.map(port => {
+      if (typeof port === 'number') {
+        return {
+          containerPort: port,
+          hostPort: port,
+        };
+      } else if (typeof port === 'string') {
+        const [containerPort, hostPort] = port.split(':');
+        return {
```

Comment:
Just wondering if you can have a string with a single port ?

Reply:
Oh yeah probably

Reply:
I think I will add slowly but surely the compose from https://github.com/docker/awesome-compose, for now I would rather keep it like that, and add more and more example to cover the features and niche cases